# RCS1236: Use exception filter

| Property                 | Value   |
| ------------------------ | ------- |
| Id                       | RCS1236 |
| Category                 | Usage   |
| Severity                 | Info    |
| Minimal Language Version | 6\.0    |

## Example

### Code with Diagnostic

```csharp
try
{
}
catch (Exception ex)
{
    if (!(ex is InvalidOperationException)) // RCS1236
    {
        throw;
    }

    return;
}
```

### Code with Fix

```csharp
try
{
}
catch (Exception ex) when (ex is InvalidOperationException)
{
    return;
}
```

## See Also

* [What's New in C# 6.0 - Exception Filters](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-6#exception-filters)
* [How to Suppress a Diagnostic](../HowToConfigureAnalyzers.md#how-to-suppress-a-diagnostic)


*\(Generated with [DotMarkdown](http://github.com/JosefPihrt/DotMarkdown)\)*